utilities_common/platform_sfputil_helper.py

```diff
@@ -131,10 +131,10 @@ def is_rj45_port(port_name):
         if not platform_porttab_mapping_read:
             platform_sfputil_read_porttab_mappings()
 
-        physical_port = logical_port_name_to_physical_port_list(port_name)[0]
         try:
+            physical_port = logical_port_name_to_physical_port_list(port_name)
             port_type = platform_chassis.get_port_or_cage_type(physical_port)
```

Comment:
This API, get_port_or_cage_type() I find is not implemented by all vendors which will result in failures and setting port_type=None. This will be an issues still.

Can we just check if this API logical_port_name_to_physical_port_list(port_name), returns back None -- if so set the port_type=None ?

Reply:
Thanks 
API get_port_or_cage_type() returning None is not an issue. In this case we just treat it as a normal xSFP port.
We expect a valid type only for RJ45 ports.
If we are confident that the API logical_port_name_to_physical_port_list will never raise an exception in any case, we can just check the return value without putting it into a try block
But I am not sure whether it’s safe enough so I suggest keeping as it is.
How do you think?
@judyjoseph

Reply:
Ok, it is fine then -- a value of None is treated as a normal xSFP port works ok.

Reply:
Another question **physicalPort** is of "list" datatype
will the API get_port_or_cage_type accept list datatype as input parameter?

Reply:
> Another question **physicalPort** is of "list" datatype will the API get_port_or_cage_type accept list datatype as input parameter?

Nice catch. Fixed and verified on both RJ45 and xSFP ports.
```
admin@mtbc-sonic-03-2700:~$ show int st
     Interface            Lanes    Speed    MTU    FEC    Alias            Vlan    Oper    Admin    Type    Asym PFC
--------------  ---------------  -------  -----  -----  -------  --------------  ------  -------  ------  ----------
     Ethernet0              0,1      50G   9100    N/A    etp1a           trunk      up       up    RJ45         off
     Ethernet2              2,3      50G   9100    N/A    etp1b           trunk      up       up    RJ45         off
     Ethernet4              4,5      50G   9100    N/A    etp2a          routed    down     down     N/A         off
     Ethernet6              6,7      50G   9100    N/A    etp2b          routed    down     down     N/A         off
```